crypto-keys/src/main/java/com/palantir/crypto2/keys/KeyStorageStrategy.java

```diff
@@ -24,15 +26,15 @@ public interface KeyStorageStrategy {
     /**
      * Stores the {@link KeyMaterial} for a file with the given {@code fileKey}.
      */
-    void put(String fileKey, KeyMaterial keyMaterial);
+    void put(String fileKey, KeyMaterial keyMaterial) throws IOException;
```

Comment:
I'd really like to avoid throwing IOExceptions to manage flow control. It seems like `remove` is the only problematic method here, so it might be worth adding a new method that differentiates between NotFound, Failed, and Succeeded?